File: apps/frontend/web-ui/src/types/notification.ts
export interface Notification {
    id: string;
    userId: string;
    type: NotificationType;
    channel: NotificationChannel;
    title: string;
    message: string;
    metadata?: any;
    status: NotificationStatus;
    createdAt: string;
    sentAt?: string;
    readAt?: string;
    read?: boolean;
  }
  
  export enum NotificationType {
    TASK_CREATED = 'TASK_CREATED',
    TASK_UPDATED = 'TASK_UPDATED',
    TASK_DELETED = 'TASK_DELETED',
    TASK_ASSIGNED = 'TASK_ASSIGNED',
    TASK_DUE_SOON = 'TASK_DUE_SOON',
    CALENDAR_EVENT_CREATED = 'CALENDAR_EVENT_CREATED',
    CALENDAR_EVENT_UPDATED = 'CALENDAR_EVENT_UPDATED',
    CALENDAR_EVENT_REMINDER = 'CALENDAR_EVENT_REMINDER',
    CALENDAR_EVENT_CANCELLED = 'CALENDAR_EVENT_CANCELLED'
  }
  
  export enum NotificationChannel {
    EMAIL = 'EMAIL',
    WEBSOCKET = 'WEBSOCKET',
    IN_APP = 'IN_APP'
  }
  
  export enum NotificationStatus {
    PENDING = 'PENDING',
    SENT = 'SENT',
    FAILED = 'FAILED',
    READ = 'READ'
  }

File: apps/backend/notification-service/src/main/java/com/agendapulse/notification/entity/Notification.java
package com.agendapulse.notification.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "notifications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private NotificationType type;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private NotificationChannel channel;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String message;

    @Column(columnDefinition = "jsonb")
    private String metadata;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private NotificationStatus status = NotificationStatus.PENDING;

    @Column(name = "created_at")
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "sent_at")
    private LocalDateTime sentAt;

    @Column(name = "read_at")
    private LocalDateTime readAt;

    public enum NotificationType {
        TASK_CREATED,
        TASK_UPDATED,
        TASK_DELETED,
        TASK_ASSIGNED,
        TASK_DUE_SOON,
        CALENDAR_EVENT_CREATED,
        CALENDAR_EVENT_UPDATED,
        CALENDAR_EVENT_REMINDER,
        CALENDAR_EVENT_CANCELLED
    }

    public enum NotificationChannel {
        EMAIL,
        WEBSOCKET,
        IN_APP
    }

    public enum NotificationStatus {
        PENDING,
        SENT,
        FAILED,
        READ
    }
}